pkg/apis/pipeline/v1beta1/resource_types.go

```diff
@@ -129,7 +134,7 @@ type PipelineResourceResult struct {
 	// The field ResourceRef should be deprecated and removed in the next API version.
 	// See https://github.com/tektoncd/pipeline/issues/2694 for more information.
 	ResourceRef *PipelineResourceRef `json:"resourceRef,omitempty"`
-	ResultType  ResultType           `json:"type,omitempty"`
+	Type        ResultType           `json:"type,omitempty"`
```

Comment:
I think it (and the other ones you commented on) should be changed - otherwise, we have to homebrew some way to ignore these specific cases of the field name not matching the JSON tag.